hook/hook.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// Package hook defines points in an SAE block's lifecycle at which common or
+// user-injected behaviour needs to be performed. Functions in this package
+// SHOULD be called by all code dealing with a block at the respective point in
+// its lifecycle, be that during validation, execution, or otherwise.
+package hook
+
+import (
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/vms/components/gas"
+	"github.com/ava-labs/libevm/core/state"
+	"github.com/ava-labs/libevm/core/types"
+	"github.com/ava-labs/libevm/params"
+
+	"github.com/ava-labs/strevm/blocks"
+	"github.com/ava-labs/strevm/gastime"
+	"github.com/ava-labs/strevm/intmath"
+	saeparams "github.com/ava-labs/strevm/params"
+)
+
+// Points define user-injected hook points.
+type Points interface {
+	GasTarget(parent *types.Block) gas.Gas
+	SubSecondBlockTime(*types.Block) gas.Gas
+	BeforeBlock(params.Rules, *state.StateDB, *types.Block) error
+	AfterBlock(*state.StateDB, *types.Block, types.Receipts)
```

Comment:
What is the rationale for allowing `BeforeBlock` to return an error but _not_ allowing `AfterBlock` to return an error?